src/main/java/io/cdap/plugin/dlp/configs/DlpFieldTransformationConfig.java

```diff
@@ -0,0 +1,191 @@
+      Arrays.stream(fields).map(field -> FieldId.newBuilder().setName(field).build()).collect(Collectors.toList())
+    );
+
+    if (fields.length == 0 || "NONE".equals(fields[0])) {
+      fieldTransformationBuilder.setPrimitiveTransformation(transformProperties.toPrimitiveTransform());
+    } else {
+
+      SensitiveDataMapping sensitivityMapping = new SensitiveDataMapping();
+      List<InfoType> sensitiveInfoTypes = sensitivityMapping.getSensitiveInfoTypes(filters);
+      InfoTypeTransformation.Builder infoTypeTransformationBuilder = InfoTypeTransformations.InfoTypeTransformation
+        .newBuilder();
+
+      infoTypeTransformationBuilder.addAllInfoTypes(sensitiveInfoTypes);
+
+      infoTypeTransformationBuilder.setPrimitiveTransformation(transformProperties.toPrimitiveTransform());
+
+      fieldTransformationBuilder.setInfoTypeTransformations(
+        InfoTypeTransformations.newBuilder().addTransformations(infoTypeTransformationBuilder));
+    }
+
+    return fieldTransformationBuilder.build();
+
+  }
+
+  /**
+   * Validates that the current set of properties in this object can form a valid FieldTransform object
+   *
+   * @param collector   FailureCollector that will be used to record all errors
+   * @param inputSchema Schema expected by the plugin as input
+   * @param widgetName  Name of the widget/field in the config that will be highlighted if an error is found
+   */
+  public void validate(FailureCollector collector, Schema inputSchema, final String widgetName) {
+    // No need to validate 'transform' field since it is used to deserialize this object
+    // So any invalid values would have been caused an error during deserialization
+
+    Gson gson = new Gson();
+    ErrorConfig errorConfig = getErrorConfig();
+    if (fields.length == 0) {
+      errorConfig.setTransformPropertyId("fields");
+      collector.addFailure(String.format("No fields were selected to apply '%s' transform.", this.transform), "")
+        .withConfigElement(widgetName, gson.toJson(errorConfig));
+    }
+
+    List<Schema.Type> supportedTypes = transformProperties.getSupportedTypes();
+
+    for (String fieldName : this.fields) {
+      Schema.Field field = inputSchema.getField(fieldName);
+      if (field == null) {
+        errorConfig.setTransformPropertyId("fields");
+        collector.addFailure(String.format("Field '%s' is not present in the input schema", fieldName), "")
+          .withConfigElement(widgetName, gson.toJson(errorConfig));
+      } else {
+
+        Schema.Type fieldType =
+          field.getSchema().isNullable() ? field.getSchema().getNonNullable().getType() : field.getSchema().getType();
+        if (!supportedTypes.contains(fieldType)) {
+          errorConfig.setTransformPropertyId("fields");
+          collector
+            .addFailure(String.format("Field '%s' has type '%s' which is not supported by '%s' transform", fieldName,
+                                      fieldType.toString(), this.transform), "")
```

Comment:
we should use display name method for schema type/logical types

Reply:
done